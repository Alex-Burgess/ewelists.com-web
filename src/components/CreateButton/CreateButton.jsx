import React from "react";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// nodejs library that concatenates classes
// @material-ui/icons
import Playlist from "@material-ui/icons/PlaylistAdd";
// @material-ui/core components
import CircularProgress from '@material-ui/core/CircularProgress';

import Button from "components/CustomButtons/Button.jsx";

import withStyles from "@material-ui/core/styles/withStyles";


import createButtonStyle from "assets/jss/material-kit-pro-react/components/createButtonStyle.jsx";

class CreateButton extends React.Component {
  render() {
    const { classes, text, isCreating, ...rest } = this.props;

    return (
        <div className={classes.addList}>
          <div>
          {!isCreating
            ? <div>
                <div className={classes.centerButton}>
                  <Button {...rest} round justIcon color="info" size="lg">
                    <Playlist />
                  </Button>
                </div>
                <p className={classes.cardDescription}>
                  {text}
                </p>
              </div>
          : <div>
              <div className={classes.centerButton}>
                <Button {...rest} round justIcon color="info" size="lg" disabled>
                  <Playlist />
                </Button>
                <CircularProgress className={classes.progress} size={72}/>
              </div>
              <p className={classes.cardDescription}>
                Creating...
              </p>
            </div>
          }
        </div>
      </div>
    );
  }
}

CreateButton.propTypes = {
  classes: PropTypes.object,
  text: PropTypes.string,
  isCreating: PropTypes.bool
};

export default withStyles(createButtonStyle)(CreateButton);
