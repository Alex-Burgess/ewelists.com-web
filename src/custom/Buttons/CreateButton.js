import React from "react";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// nodejs library that concatenates classes
// @material-ui/icons
import Playlist from "@material-ui/icons/PlaylistAdd";
// @material-ui/core components
import CircularProgress from '@material-ui/core/CircularProgress';
import Button from "components/CustomButtons/Button.js";
import { makeStyles } from "@material-ui/core/styles";


import styles from "assets/jss/custom/components/createButtonStyle.js";
const useStyles = makeStyles(styles);


export default function CreateButton(props) {
  const { text, isCreating, ...rest } = props;
  const classes = useStyles();

  return (
      <div className={classes.addList}>
        <div>
        {!isCreating
          ? <div>
              <div className={classes.centerButton}>
                <Button {...rest} round justIcon color="info" size="lg">
                  <Playlist />
                </Button>
              </div>
              <p className={classes.cardDescription}>
                {text}
              </p>
            </div>
        : <div>
            <div className={classes.centerButton}>
              <Button {...rest} round justIcon color="info" size="lg" disabled>
                <Playlist />
              </Button>
              <CircularProgress className={classes.progress} size={72}/>
            </div>
            <p className={classes.cardDescription}>
              Creating...
            </p>
          </div>
        }
      </div>
    </div>
  );
}

CreateButton.propTypes = {
  text: PropTypes.string,
  isCreating: PropTypes.bool
};
