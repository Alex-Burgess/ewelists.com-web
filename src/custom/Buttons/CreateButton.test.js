import * as React from 'react';
import renderer from 'react-test-renderer';
import CreateButton from './CreateButton';

test('renders amazon button', () => {
  const tree = renderer
    .create(
      <CreateButton
        text="Create a New List"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
