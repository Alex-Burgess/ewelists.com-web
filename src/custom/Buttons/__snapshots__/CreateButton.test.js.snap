// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders amazon button 1`] = `
<div
  className="makeStyles-addList-1"
>
  <div>
    <div>
      <div
        className="makeStyles-centerButton-3"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-5 makeStyles-lg-30 makeStyles-info-9 makeStyles-round-32 makeStyles-justIcon-35"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M14 10H2v2h12v-2zm0-4H2v2h12V6zm4 8v-4h-2v4h-4v2h4v4h2v-4h4v-2h-4zM2 16h8v-2H2v2z"
              />
            </svg>
          </span>
        </button>
      </div>
      <p
        className="makeStyles-cardDescription-2"
      >
        Create a New List
      </p>
    </div>
  </div>
</div>
`;
